src/main/java/com/google/maps/DirectionsClientImpl.java

```diff
@@ -46,38 +39,34 @@ public DirectionsClient getDirectionsClient(String apiKey) {
     }
   }
 
-  /**
-   * Gets the result of a GET request to the Google Directions API.
-   *
-   * @param destination A string representing the destination to get directions to.
-   * @param origin A string representing the origin to get directions from.
-   * @param waypoints A list of string consisting of waypoints to visit between the destination and
-   *     the origin.
-   * @return A string representing the result from a GET request to the Google Directions API.
-   * @throws Exception
-   */
-  public List<String> getDirections(String origin, String destination, String[] waypoints)
-      throws ApiException, InterruptedException, IOException {
+  @Override
+  public List<String> getDirections(String origin, String destination, List<String> waypoints)
+      throws DirectionsException {
+    String[] waypointsArray = waypoints.toArray(new String[0]);
     try {
       DirectionsResult result =
           directionsService
               .origin(origin)
               .destination(destination)
-              .waypoints(waypoints)
+              .waypoints(waypointsArray)
               .optimizeWaypoints(true)
               .await();
-      List<String> listLegs = new ArrayList<>();
-      DirectionsRoute routes[] = result.routes;
-      for (DirectionsRoute route : routes) {
-        DirectionsLeg legs[] = route.legs;
-        for (DirectionsLeg leg : legs) {
-          listLegs.add(leg.toString());
-        }
-      }
-      return listLegs;
+
+      return Arrays.asList(result.routes).stream()
+          .map(
+              route -> {
+                return Arrays.asList(route.legs).stream()
+                    .map(
+                        leg -> {
+                          return leg.toString();
+                        })
+                    .collect(Collectors.toList());
```

Comment:
Is this really necessary? I wonder if Java can handle the stream directly

Reply:
If I remove `.collect(Collectors.toList());` and `.flatMap(Collection::stream);`, it would return a `List<Stream<String>>` instead of a `List<String>`. I am not sure how to append the streams together and make that into list.

Reply:
Gotcha. I'm not if there is a simpler way, but it looks good now

Reply:
Nice, thanks!

Reply:
`RESOLVED`